app/controllers/posts_controller.rb

```diff
@@ -98,32 +101,18 @@ def fetch_post(viewed_by_seller = true)
         e404
       end
       e404 if @post.blank?
-
-      if viewed_by_seller
-        e404 if @post.seller != current_seller
-      end
-
-      if @post.seller_id?
-        e404 if @post.seller.suspended?
-      elsif @post.link_id?
-        e404 if @post.link.seller&.suspended?
-      end
     end
 
     def check_if_needs_redirect
-      fetch_post(false)
```

Comment:
Part of re-factor from https://github.com/antiwork/gumroad/pull/2246#discussion_r2599264294

Post will always be present because of  `before_action` callback excecution order